service/go

```diff
@@ -1,167 +1,30 @@
-        echo "Executing JSHint."
-        cd ../web-ui
-        LC_ALL=en_US.UTF-8 ./go jshint
-        cd -
-        echo "Done."
-}
-
-function runCoverageUnit {
-        echo "Generating Unit Test Converage Information."
-        coverage erase
-        getTrialAbsolutePath
-        coverage run -p --source=pixelated $TRIAL_PATH --reporter=$TRIAL_REPORTER $* test.unit
-        coverage combine
-        coverage html
-        echo "Done."
-}
-
-function runCoverageIntegration {
-        echo "Generating Integration Test Converage Information."
-        coverage erase
-        getTrialAbsolutePath
-        coverage run -p --source=pixelated $TRIAL_PATH --reporter=$TRIAL_REPORTER $* test.integration
-        coverage combine
-        coverage html
-        echo "Done."
-}
-
-function runCoverageUnitAndIntegration {
-        echo "Generating Unit and Integration Test Converage Information."
-        coverage erase
-        getTrialAbsolutePath
-        coverage run -p --source=pixelated $TRIAL_PATH --reporter=$TRIAL_REPORTER test.unit
-        coverage run -p --source=pixelated $TRIAL_PATH --reporter=$TRIAL_REPORTER test.integration
-        coverage combine
-        coverage html
-        echo "Done."
-}
-
-function runFunctionalTests {
-        echo "Executing Functional Tests on headless PhantomJS."
-        removeZmqCertificates
-        echo "You should execute it on Debian box for more similar results with CI environment."
-        behave --tags ~@wip --tags ~@smoke test/functional/features
-        echo "Done."
-}
-
-function cleanPyc {
-    find . -name '*.pyc'  -delete
-}
-
 if [ "$1" == 'test' ]; then
         set -e
-        runPep8
-        runUnitTests "${@:2}"
-        runIntegrationTests "${@:2}"
-elif [ "$1" == 'unit' ]; then
-        set -e
-        cleanPyc
-        runUnitTests
-elif [ "$1" == 'integration' ]; then
-        set -e
-        cleanPyc
-        runIntegrationTests
+        pushd ..
+        make test_py
+        popd
 elif [ "$1" == 'pep8' ]; then
         set -e
-        runPep8
-elif [ "$1" == 'setuppy' ]; then
-        setuppy
-elif [ "$1" == 'setupjs' ]; then
-        setupjs
+        pushd ..
+        make linters_js
+        popd
 elif [ "$1" == 'setup' ]; then
-        setupjs
-        setuppy "${@:2}"
-elif [ "$1" == 'coverage_unit' ]; then
-        runCoverageUnit "${@:2}"
-elif [ "$1" == 'coverage_integration' ]; then
-        runCoverageIntegration "${@:2}"
+        set -e
+        pushd ..
+        make setup
+        popd
 elif [ "$1" == 'coverage_all' ]; then
         set -e
-        runPep8
-        runCoverageUnitAndIntegration "${@:2}"
-elif [ "$1" == 'start' ]; then
-        /usr/bin/env pixelated-user-agent "${@:2}"
+        pushd ..
+        make coverage
+        popd
 elif [ "$1" == "functional" ]; then
-        runFunctionalTests "${@:2}"
+        set -e
+        pushd ..
+        make test_functional
+        popd
 else
-        python setup.py $*
+        echo "Command doesn't exist"
```

Comment:
`make` will throw an error if it doesn't find the command, as far as I know. You don't have to implement it in `go` as well. Or you can take the response from make and add the `exit 1`, but it might not be necessary. But I'd say having the huge `if` will be really tough if in the future we want to add a command to the makefile... then we have to maintain two different files for running these commands. Does that make sense?